src/components/organisms/Top.tsx

```diff
@@ -2,24 +2,51 @@ import React, { useState, useEffect } from "react";
 import styles from "@/styles/components/organisms/Top.module.scss";
 import Weather from "@/components/molecules/Weather";
 import Livedata from "./Livedata";
+import axios from "axios";
+
+export type WeatherData = {
+  location_name: string;
+  temperature_c: number;
+  temperature_f: number;
+  humidity: number;
+  weather_icon: string;
+};
 
 export default function Top() {
-  const [lat, setLat] = useState<number | null>();
-  const [long, setLong] = useState<number | null>();
   const [currTemp, setCurrTemp] = useState<number | null>();
   const [currHumid, setCurrHumid] = useState<number | null>();
+  const [deviceId, setDeviceId] = useState<string | null>();
+  const [weatherData, setWeatherData] = useState<WeatherData | []>([]);
 
```

Comment:
Is the default value need to be `[]`? `weatherData` will be a single data not a array. Isn't `null`?